tests/testthat/test-s3.R

```diff
@@ -65,13 +65,22 @@ test_that("S3 methods of Zarr object", {
 
     #subetting with brackets `[`
     expect_equal(z[4,5], z$get_item(list(slice(4, 4), slice(5, 5))))
-    expect_equal(z[1:3,5], z$get_item(list(slice(1, 3), slice(5, 5))))
-    #TODO slicing by step size is not yet supported
-    expect_error(z[1,seq(1,5,2)])
+    
+    # with seq
+    # for now seq(from, to, by) does not use slice(start,step,steo)
+    z[1:2,seq(1,5,2)]
+
+    # arbitrary indices
+    z[c(2,1),c(10,1,6)]
+      
+    # compare regular slicing and step size
+    expect_error(z[1], "This Zarr object has 2 dimensions, 1 were supplied")
+    
+    # test `[.ZarrArray` against get_item
+    expect_equal(z[1:2,5], z$get_item(list(slice(1, 3), slice(5, 5))))
```

Comment:
It looks like this line was previously 
```r
expect_equal(z[1:3,5], z$get_item(list(slice(1, 3), slice(5, 5))))
```
Did the behavior change?

Reply:
Any ideas ?

Reply:
We are almost ready :D please let me know how we should treat `subscript out of bounds` cases like descibed above.